swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/Store.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  Store.swift
+//  (c) 2024 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import Combine
+import NetworkExtension
+import UserNotifications
+import OSLog
+
+#if os(macOS)
+  import AppKit
+#endif
+
+@MainActor
+public final class Store: ObservableObject {
+  @Published private(set) var actorName: String?
+
+  // Make our tunnel configuration convenient for SettingsView to consume
+  @Published private(set) var settings: Settings
+
+  // Enacapsulate Tunnel status here to make it easier for other components
+  // to observe
+  @Published private(set) var status: NEVPNStatus
+
+  // This is not currently updated after it is initialized, but
+  // we could periodically update it if we need to.
+  @Published private(set) var decision: UNAuthorizationStatus
```

Comment:
All of the `@Published` properties here are used for bindings in views, so it was appropriate to add one for the Notification decision (iOS).

In reality that won't ever change after the user makes a selection, but it's here for completeness.